collectors/modality-probe-debug-collector/src/main.rs

```diff
@@ -0,0 +1,22 @@
+use std::sync::mpsc::channel;
+use structopt::StructOpt;
+
+mod cli;
+use cli::{config_from_options, CLIOptions};
+
+use modality_probe_debug_collector::run;
+
+fn main() {
+    let opts = CLIOptions::from_args();
+    let config = match config_from_options(opts) {
+        Ok(cfg) => cfg,
+        Err(err) => {
+            println!("{}", err);
+            return;
+        }
+    };
+    let (_, shutdown_receiver) = channel();
```

Comment:
It looks like the piping is incomplete here, since there's nothing holding on to the shutdown channel's sender.  I presume this is supposed to be hooked up to a control-C type hook.